### DISCUSSION TITLE
Console example compatible with ShapesDemo

### DISCUSSION THREAD

**Discussion by Johnsik** [eProsima/Fast-DDS#4115] [Q&A] [answered]
Is there some simple console example for which performs functions similar to the Publisher in ShapesDemo?
Even without settings.

**Comment by JesusPoderoso** [comment, marked as answer]
Hi @Johnsik, thanks for using Fast DDS.
I am afraid there is no such example you are describing.
In case you want to build it manually, the following snippet is a ShapesDemo IDL example. You can [generate the types with Fast DDS-Gen](https://fast-dds.docs.eprosima.com/en/latest/fastddsgen/pubsub_app/pubsub_app.html), and use it with a publisher - subscriber as the [Helloworld example](https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/dds/HelloWorldExample).
```
struct ShapeType
{
	@Key string color;
	long x;
	long y;
	long shapesize;
};
```
We are open to contributions, so if you are interested and you implement it, you can create a Pull Request to Fast DDS with your implementation!
